icepyx/core/types.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from typing import Literal, TypedDict
+from typing_extensions import NotRequired
+
+
+CMRParamsBase = TypedDict(
+    "CMRParamsBase",
+    {
+        "temporal": NotRequired[str],
+        "options[readable_granule_name][pattern]": NotRequired[str],
+        "options[spatial][or]": NotRequired[str],
+        "readable_granule_name[]": NotRequired[str],
+    },
+)
+
+
+class CMRParamsWithBbox(CMRParamsBase):
+    bounding_box: str
+
+
+class CMRParamsWithPolygon(CMRParamsBase):
+    polygon: str
+
+
+CMRParams = CMRParamsWithBbox | CMRParamsWithPolygon
+
+
+class EGISpecificParamsBase(TypedDict):
+    """Common parameters for searching, ordering, or downloading from EGI.
+
+    See: https://wiki.earthdata.nasa.gov/display/SDPSDOCS/EGI+Programmatic+Access+Documentation
+
+    EGI shares parameters with CMR, so this data is used in conjunction with CMRParams
+    to build EGI requests.
+
+    TODO: Validate more strongly (with Pydantic and its annotated types?
+    https://docs.pydantic.dev/latest/concepts/types/#composing-types-via-annotated):
+
+    * short_name is `ATL##` (or Literal list of values?)
+    * version is 1-3 digits
```

Comment:
there's a list of valid products in `is2ref._validate_product`. Some have the format `ATL##QL`.

I believe the version is always 3 digits (e.g. `006` vs `6`).

Reply:
Used that explicit list to exactly type `short_name`: 0440fa9